lib/Acho.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+const CONST = require('./Constants')
+const DEFAULT = require('./Default')
+
+function Acho (opts = {}) {
+  !(this instanceof Acho) && new Acho(opts)
+  const acho = Object.assign({ diff: [] }, DEFAULT(), opts)
+
+  acho.messages = (() =>
+    Object.keys(acho.types).reduce((messages, type) => {
+      acho[type] = acho.generateTypeMessage(type)
+      messages[type] = (opts.messages || {})[type] || []
+      return messages
+    }, {}))()
+
+  acho.push = function (type, ...messages) {
```

Comment:
why not fat arrow for bind the context?

Reply:
Because wherever you need to use this you need to use a regular function, arrows don't work here